apps/ledger-live-mobile/src/components/DeviceAction/index.tsx

```diff
@@ -35,29 +35,52 @@ type Props<R, H, P> = {
   onResult?: (_: any) => Promise<void> | void;
   onError?: (_: any) => Promise<void> | void;
   renderOnResult?: (_: P) => React.ReactNode;
-  action: Action<R, H, P>;
+  action?: Action<R, H, P>;
+  status?: any;
   request?: R;
   device: Device;
+  payload?: any;
   onSelectDeviceLink?: () => void;
   analyticsPropertyFlow?: string;
 };
+
 export default function DeviceAction<R, H, P>({
   action,
   request = null,
   device: selectedDevice,
+  ...props
+}: Props<R, H, P>): JSX.Element {
+  const status: any = action.useHook(selectedDevice, request);
+  const payload = action.mapResult(status);
```

Comment:
To make TS a bit happier: `action?.useHook` and `action?.mapResult` because action could be `undefined` ;)